interfaces/dns_record/v0/README.md

```diff
@@ -0,0 +1,107 @@
+# `dns_record`
+
+## Usage
+
+This relation interface describes the expected behavior of any charm claiming to be able to interface with a DNS record Provider.
+
+## Direction
+
+The `dns_record` interface implements a provider/requirer pattern. The requirer is a charm that wishes to create a set of DNS records, and the provider is the charm managing those..
+```mermaid
+flowchart TD
+  Requirer -- dns-domains, dns-entries --> Provider
+  Provider -- dns-domains, dns-entries --> Requirer
+```
+
+## Behavior
+
+The following is the criteria that a Provider and Requirer need to adhere to be compatible with this interface.
+
+### Provider
+
+- Is expected to provide a list of dns-domains and/or a list of dns-entries in the relation databag, each containing the domain, the status and optionally the status_description.
+
+### Requirer
+
+- Is expected to provide a list of dns-domains and/or a list of dns-entries in the relation databag, each containing the domain it is requesting a DNS record for, the username and the password containing a juju secret to authenticate the request.
```

Comment:
It's not clear reading this what the username/passwords are for. I'm aware this is sort of explained in the spec but specs aren't public and the charm relation interface content here will be rendered in Charmhub so it's important to place any context specific information here.

Reply:
IMO, "authenticate the request" is already clear. What else would you expect?

Reply:
Well, as I see it, the requirer is agnostic to the provider internals. I've updated the description, though. Perhaps now it's a bit clearer

Reply:
Perhaps we should say (for the provider) "The provider should authenticate requests for dns-domains based on internal business rules/processes at the organisation where this charm is deployed."?

Reply:
Made some changes based on the feedback. Feel free to reopen if you still think it's unclear

Reply:
Sorry for insisting here, it's still unclear to me, reading this doc, how the requirer can know about the username/password of the provider. It seems the username/passwords are managed by the provider app, so why is the requirer asked to pass them over? Does this mean that every DNS requirer will have config (or juju secrets) that the user will set to their credentials?

I guess the general problem arises from the fact that DNS (and networking in general) don't really fit well with the requirer/provider pattern, it's more of an underlying substrate that charms use.

Reply:
@gruyaume this is similar to how you might need to know access credentials for interacting with AWS or how you might need to know a token for using Ubuntu Pro, for instance. We're designing this for a situation where one team is running the DNS provider (an IT department, for instance) and many other teams are deploying and managing applications that might want to create and manage DNS records, such as development teams. Not all development teams should have access to write any DNS record for the organisation in question, so we need a mechanism for doing that. The credentials need to be negotiated outside of the charm relation. Does that make sense? Happy to jump on a call to discuss if this isn't clear.

Reply:
That sounds like a good approach, to add that wording to the provider portion of the README. So something like:
```
- Is expected to authenticate requests for dns-domains based on internal business rules/processes at the organisation where this charm is deployed. The credentials need to be negotiated outside of the charm relation.
```

Reply:
Updated